### PR TITLE
included the readme for both worker and scheduler, included proto file and a makefile

### PR DESCRIPTION
**Pull request by KorayGocmen**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1 +1,59 @@
...
+
+
+# Goal
+
+*   Build a scheduler and worker architecture that supports GRPC communication.
+
+# General Overview
+
+*   Support single scheduler and multiple workers.
+*   The scheduler must be aware of all workers' states. (requires the workers to register and deregister when going offline)
```
Comment (knisbet) [inline readme.md:11]: Do you think it can be guaranteed, that the workers will properly deregister when going offline?

**Review by knisbet — APPROVED** [review]
Nothing stands out so far, I'll be interested to see how this develops, and see what the security model looks like.

File: readme.md (hunk 1 above)
Comment (KorayGocmen) [reply readme.md:11]: Yes, as long as there wasnt an unexpected error. I will implement a fatal function that calls the deregister function before killing the app

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+SCHEDULER_BINARY = scheduler
+
+WORKER_PATH = worker
+WORKER_BINARY = worker
+
+PROTO_PATH = jobscheduler
+PROTO_NAME = job_scheduler.proto
+PROTO_PLUGIN = jobscheduler
+
+all: clean build
```
Comment (klizhentas) [inline Makefile:13]: Nitpick: those targets are .PHONY

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+build_scheduler:
+	go build -o $(SCHEDULER_PATH)/$(SCHEDULER_BINARY) $(SCHEDULER_PATH)/*.go
+
+build_worker:
+	go build -o $(WORKER_PATH)/$(WORKER_BINARY) $(WORKER_PATH)/*.go
+
+clean: 
+	rm -f *.out
+	rm -f $(SCHEDULER_PATH)/$(SCHEDULER_BINARY)
+	rm -f $(WORKER_PATH)/$(WORKER_BINARY)
```
Comment (klizhentas) [inline Makefile]: missing newline

File: jobscheduler/job_scheduler.proto [hunk 4]
```diff
@@ -0,0 +1,72 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "com.gravitational.jobscheduler";
+option java_outer_classname = "JobSchedulerServerProto";
+
+package jobscheduler;
+
+// Scheduler server methods
```
Comment (klizhentas) [inline jobscheduler/job_scheduler.proto:9]: Hint: for final version we would ask for more detailed comments according to https://github.com/golang/go/wiki/CodeReviewComments

File: readme.md [hunk 5]
```diff
@@ -1 +1,59 @@
...
+*   The scheduler and the workers authenticate with each other via SSL.
```
Comment (klizhentas) [inline readme.md]: NIt: these days it's TLS, as SSL is dated version

File: readme.md [hunk 6]
```diff
@@ -1 +1,59 @@
...
+        *   Stop a job
+        *   Return the status of a job
+        *   Return a stream of output for a running job.
+
+
+# Scheduler Overview
+
+* [Scheduler Details](scheduler/scheduler.md)
+* All config parameters are specified in the config.toml file
+* When a worker registers, a UUID is assigned to the worker and worker details are kept in a map.
```
Comment (klizhentas) [inline readme.md:28]: You are using in-memory model, which is fine. Please write down (but don't implement) some notes on what backend do you want to pick for persistence and why.

File: readme.md [hunk 7]
```diff
@@ -1 +1,59 @@
...
+
+# Worker Overview
+
+* [Worker Details](worker/worker.md)
+* All config parameters are specified in the config.toml file
+* Support a GRPC-API (Appendix B)
+* Jobs are basically scripts that are held in the specified folder in the config.
+* When a job is started by the scheduler, a job object is created by the worker
+  * This object specifies where the output of the job will be piped.
+  * Also holds which command and which path was requested.
```
Comment (klizhentas) [inline readme.md]: Write down a section on design tradeoffs and high availability trade offs and rationale.

**Pull request merged**